### DISCUSSION TITLE
Documentation: flows

### DISCUSSION THREAD

**Discussion by Foxite** [goauthentik/authentik#3044] [Q&A] [answered]
I am thoroughly confused by flows and how to create new ones. I wanted to test invitations but there was a warning that there are no invitiation stages bound to any flows. I have examined the default enrollment flow and I have no clue where to place an invitation stage, if I should even add it to the default one or if I should create a new flow, or how the default enrollment even works at all.
- The diagram shows a policy `default-source-enrollment-if-username`, but this policy is not mentioned in the configuration for the flow. Why is it there?
- The diagram shows specific connections between policies and subsequent stages, but I do not see this being reflected in the configuration. How is this configured?
- Am I missing documentation resources for flows? The size of [this page](https://goauthentik.io/docs/flow/) is tiny compared to how powerful flows seem to be.
- Is there an example flow for invitations?

**Comment by unixfy** [comment, marked as answer]
To answer some of your questions:

- I agree that the "invitation stage missing" thing is a bit of a misnomer. Actually, both of the options you suggested will work. You can either create a new enrollment flow with an invitation stage, or add an invitation stage to the existing default enrollment flow. If you choose the former, you'd be able to enroll users (create users) both with and without an invitation depending on which flow you want to use. The invitation stage should go before the "default user registration" stage so users can't attempt to create an account unless they have an invitation.
- If I remember correctly this policy `default-source-enrollment-if-username` is designed to prevent the enrollment flow from triggering if the user tries to create an account by signing in using an OAuth source (i.e., social login like GitHub, Discord, Facebook...). That way the user isn't prompted to choose their own username; authentik will just use the username provided by the OAuth IDP.
- You can bind policies by going to Flows & stages > Flows > Select a flow > Policy / group / user bindings > Create binding and select a policy.
- No, documentation for flows is pretty limited. PRs always welcome 🙏
- The default enrollment stage with an "invitation" stage right before the first "prompt" stage will work fine.

Happy to help if you've got further questions!

**Reply by Foxite** [reply]
I was able to get invitations working with your help, thanks!

To clarify, I saw the `default-source-enrollment-if-username` policy in the diagram without it being listed in the flow's policies, but I have found out that this policy is added as part of one of the stages. This also clarifies the diagram for the default source enrollment flow, because it looked pretty wild.

Shall I mark this as the answer or should I leave this open to appeal for documentation PRs?

**Reply by unixfy** [reply]
Great! I'm glad to hear that you got it sorted.

> Shall I mark this as the answer or should I leave this open to appeal for documentation PRs?

Since it's a Q&A and your question is resolved, marking this as the answer would work. Opening a new issue or non-Q&A discussion to improve docs for flows would probably help more people get eyes on the idea ;)
